drivers/pwm/pwm_nrfx.c

```diff
@@ -35,6 +38,26 @@ LOG_MODULE_REGISTER(pwm_nrfx, CONFIG_PWM_LOG_LEVEL);
 #define ANOMALY_109_EGU_IRQ_CONNECT(idx)
 #endif
 
+#define PWM(dev_idx) DT_NODELABEL(pwm##dev_idx)
+#define PWM_PROP(dev_idx, prop) DT_PROP(PWM(dev_idx), prop)
+#define PWM_HAS_PROP(idx, prop) DT_NODE_HAS_PROP(PWM(idx), prop)
+
+#define PWM_NRFX_IS_FAST(unused, prefix, idx, _)					\
+	COND_CODE_1(DT_NODE_HAS_STATUS_OKAY(PWM(idx)),					\
+		(COND_CODE_1(PWM_HAS_PROP(idx, power_domains),				\
+		    (IS_EQ(DT_PHA(PWM(idx), power_domains, id), NRF_GPD_FAST_ACTIVE1)),	\
+		    (0))), (0))
+
+#if NRFX_FOREACH_PRESENT(PWM, PWM_NRFX_IS_FAST, (||), (0))
+#if CONFIG_CLOCK_CONTROL
+#define PWM_NRFX_USE_CLOCK_CONTROL 1
+#endif
+#define PWM_NRFX_INIT_PRIORITY 99
+#else
+#define PWM_NRFX_INIT_PRIORITY CONFIG_PWM_INIT_PRIORITY
```

Comment:
This should be handled in Kconfig, otherwise users can be surprised that the value they set is not applied.